### PR TITLE
kubeadm: sort the merged env vars and improve related tests

### PR DESCRIPTION
**Pull request by neolit123**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

contains two commits:
```
kubeadm: add unit test for GetProxyEnvVars

kubeadm: sort the results of MergeKubeadmEnvVars 
```

@[neolit123](https://github.com/kubernetes/kubernetes/commits?author=neolit123)
neolit123 committed 3 minutes ago
[kubeadm: sort the results of MergeKubeadmEnvVars](https://github.com/kubernetes/kubernetes/commit/d899d9049574e63c9e9cf07e84274a0f502fb469)

#### Which issue(s) this PR fixes:

NONE

#### Does this PR introduce a user-facing change?

```release-note
kubeadm: make sure the extra environment variables written to a kubeadm managed PodSpec are sorted alpha-numerically by the environment variable name.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/util/env.go [hunk 1]
```diff
@@ -26,9 +27,12 @@ import (
 )
 
 // GetProxyEnvVars builds a list of environment variables in order to use the right proxy
-func GetProxyEnvVars() []kubeadmapi.EnvVar {
+func GetProxyEnvVars(environment []string) []kubeadmapi.EnvVar {
```
Comment (pacoxu) [inline cmd/kubeadm/app/util/env.go:30]: All `GetProxyEnvVars` that are used above, are `GetProxyEnvVars(nil)`.

`environment []string` is only used for testing. Can we make it clear?

**Review by pacoxu — COMMENTED** [review]
Other lgtm.

File: cmd/kubeadm/app/util/env.go (hunk 1 above)
Comment (neolit123) [reply cmd/kubeadm/app/util/env.go:31]: added comment above the function.

**Comment by pacoxu** [comment]
/hold
in case other reviewer want to take a look

**Review by SataQiu — COMMENTED** [review]
/lgtm

**Pull request merged**